pkg/store/prometheus.go

```diff
@@ -290,19 +289,18 @@ func (p *PrometheusStore) queryPrometheus(
 	}
 
 	externalLbls := rmLabels(p.externalLabelsFn().Copy(), extLsetToRemove)
+	b := labels.NewScratchBuilder(16)
 	for _, vector := range matrix {
-		seriesLbls := labels.Labels(make([]labels.Label, 0, len(vector.Metric)))
+		b.Reset()
 
 		// Attach labels from samples.
 		for k, v := range vector.Metric {
-			seriesLbls = append(seriesLbls, labels.FromStrings(string(k), string(v))...)
+			b.Add(string(k), string(v))
 		}
-		sort.Slice(seriesLbls, func(i, j int) bool {
-			return seriesLbls.Less(i, j)
-		})
-		// Attach external labels for compatibility with remote read.
-		finalLbls := labelpb.ExtendSortedLabels(seriesLbls, externalLbls)
-		finalLbls = append(finalLbls, dedup.PushdownMarker)
+		b.Add(dedup.PushdownMarker.Name, dedup.PushdownMarker.Value)
+		b.Sort()
```

Comment:
Do we need this manual sort for the builder? Thought it is sorted.

Reply:
Builder is sorted, Scratchbuilder is not!